File: An2/Sem1/MAP/MAP-C#/MAP-Lab9/MAP-Lab9/Controller/Validator.cs
using MAP_Lab9.Model;
using MAP_Lab9.Repository;
using MAP_Lab9.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Lab9.Controller
{
    class Validator
    {
        private Repository<Student> repo;

        public Validator(Repository<Student> repo)
        {
            this.repo = repo;
        }

        private bool studentExists(int id)
        {
            IDictionary<int, Student> students = repo.getAllElements();

            foreach (Student student in students.Values)
                if (student.getID() == id)
                    return true;

            return false;
        }

        private bool nameIsEmpty(String name)
        {
            if (name.Length < 1) return true;
            else return false;
        }

        public void validateStudent(Student s)
        {
            if (this.nameIsEmpty(s.getName()))
                throw new MyException("Name cannot be empty!\n");
            if (this.studentExists(s.getID()))
                throw new MyException("Student id already exists!\n");
        }
    }
}

File: An2/Sem1/MAP/MAP-Java/MAP-Lab10/src/Controller/Validator.java
package Controller;

import Model.Student;
import Repository.Repository;
import Utils.MyException;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 12/5/13
 * Time: 2:41 PM
 * To change this template use File | Settings | File Templates.
 */
public class Validator {

    private Repository repo;

    public Validator(Repository repo){
        this.repo = repo;
    }

    private boolean studentsExists(int id){
        return repo.getAllElements().containsKey(id);
    }

    private boolean nameIsEmpty(String name){
        return name.length() == 0;
    }

    public void validateStudent(Student s) throws MyException {
        if (nameIsEmpty(s.getName()))
            throw new MyException("Name cannot be empty!");
        if (studentsExists(s.getID()))
            throw new MyException("Student id already exists!");
    }

}
